src/firefly/js/ui/JoyToMUIThemeWrapper.jsx

```diff
@@ -0,0 +1,41 @@
+import React from 'react';
+import {CssVarsProvider, extendTheme as extendJoyTheme, THEME_ID, useColorScheme} from '@mui/joy/styles';
+import {useTheme as useMaterialTheme} from '@mui/material/styles';
+import {
+    Experimental_CssVarsProvider as MaterialCssVarsProvider,
+    useColorScheme as useMaterialColorScheme
+} from '@mui/material/styles/CssVarsProvider';
+
+
+/**
+ * This is needed to use MUI X components in JoyUI, otherwise they will throw errors because they expect MUI theme.
+ * Adapted from the example code present at https://mui.com/x/react-date-pickers/custom-field/#using-another-input
+ *
+ * TODO: Override the default MUI appearance of wrapped component (esp font, color, etc.) with JoyUI theme.
```

Comment:
You followed the documentation and know to create a generic wrapper, which is the correct approach.  Good work.

Reply:
Oh thanks! I was actually asking about the "TODO" - do you know how to override the theme variables (like font, color, etc.) using `CssVarsProvider`? (because if you notice, the picker renders in MUI theme, not JoyUI's theme)

Reply:
I looked into this and there does not seem to be an easy, global setting for it.  I was able to take JoyUI's font by doing this.  If this is the only way, then you can inherit color and other things in a similar fashion, I am guessing.
```js
    const theme = materialExtendTheme({
        typography: {
            fontFamily: joyTheme.fontFamily
        },
    });

    return (
        <MaterialCssVarsProvider theme={theme}>
```

Reply:
BTW, did you confirm that `SyncThemeMode` is working/needed?  It looks like it's watching MUI's color mode.  Shouldn't it be watching JoyUI's mode changes?

Reply:
Thanks - I was able to make it work! And good catch, mode changing was wrong (and even buggy), I fixed that. I rebuilt the build if you want to test again.